File: app/src/main/java/de/bornholdtlee/defaultproject/utils/FileUtils.java
package de.bornholdtlee.defaultproject.utils;

import android.content.Context;

import java.io.IOException;
import java.io.InputStream;

import de.bornholdtlee.defaultproject.Constants;

public class FileUtils {

    public String getJsonFromFile(Context context, String filePath) {
        String json = "";
        try {
            InputStream inputStream = context.getAssets().open(filePath);
            int size = inputStream.available();
            byte[] buffer = new byte[size];
            inputStream.read(buffer);
            inputStream.close();
            json = new String(buffer, Constants.ENCODING_UTF8);
            json = json.replace("\n", "");
        } catch (IOException e) {
            Logger.error("File corrupted");
        }
        return json;
    }
}

File: app/src/main/java/de/bornholdtlee/defaultproject/utils/FileUtils.kt
package de.bornholdtlee.defaultproject.utils

import android.content.Context

import java.io.IOException
import java.io.InputStream

import de.bornholdtlee.defaultproject.ENCODING_UTF8

class FileUtils {

    fun getJsonFromFile(context: Context, filePath: String): String {
        var json = ""
        try {
            val inputStream = context.assets.open(filePath)
            val size = inputStream.available()
            val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()
            json = String(buffer, ENCODING_UTF8)
            json = json.replace("\n", "")
        } catch (e: IOException) {
            Logger.error("File corrupted")
        }

        return json
    }
}
